gcsweb/cmd/gcsweb/gcsweb.go

```diff
@@ -107,14 +111,49 @@ func (o *options) validate() error {
 	if _, err := os.Stat(o.flStyles); os.IsNotExist(err) {
 		return fmt.Errorf("styles path '%s' doesn't exists.", o.flStyles)
 	}
+	if o.oauthTokenFile != "" && o.gcsCredentialsFile != "" {
+		return errors.New("specifying both --oauth-token-file and --gcs-credentials-file is not allowed.")
+	}
+
 	if o.oauthTokenFile != "" {
 		if _, err := os.Stat(o.oauthTokenFile); os.IsNotExist(err) {
 			return fmt.Errorf("oauth token file '%s' doesn't exists.", o.oauthTokenFile)
 		}
 	}
+
+	if o.gcsCredentialsFile != "" {
+		if _, err := os.Stat(o.gcsCredentialsFile); os.IsNotExist(err) {
+			return fmt.Errorf("gcs service account crendentials file '%s' doesn't exists.", o.gcsCredentialsFile)
+		}
+	}
+
 	return nil
 }
 
+func getStorageClient(o options) (*storage.Client, error) {
+	ctx := context.Background()
+	clientOption := option.WithoutAuthentication()
+
+	if o.oauthTokenFile != "" {
+		secretAgent := &secret.Agent{}
+		if err := secretAgent.Start([]string{o.oauthTokenFile}); err != nil {
+			return nil, fmt.Errorf("Error starting secrets agent: %v", err)
+		}
+		clientOption = option.WithAPIKey(string(secretAgent.GetSecret(o.oauthTokenFile)))
```

Comment:
I don't think there is any value in using the secret agent if you only actually load the secret once while initializing.

Reply:
Addressed. I removed the secret agent. Let's just read the file contents.